app/controllers/api/v2/vcs_clone_controller.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    class VcsCloneController < ::Api::V2::BaseController
+      include ::ForemanAnsible::ProxyAPI
+
+      rescue_from ActionController::ParameterMissing do |e|
+        render json: { 'error' => e.message }, status: :bad_request
+      end
+
+      rescue_from Git::GitExecuteError do |_e|
+        head :internal_server_error
+      end
+
+      api :GET, '/vcs_clone/get_repo_info', N_('Returns information about the repo')
```

Comment:
REST encodes the method in the HTTP method, so `get_repo_info` is redundant. See further suggestions in the routes part.